devito/ir/iet/nodes.py

```diff
@@ -961,12 +968,13 @@ class Section(List):
 
     is_Section = True
 
-    def __init__(self, name, body=None):
+    def __init__(self, name, body=None, is_subsection=False):
         super(Section, self).__init__(body=body)
         self.name = name
+        self.is_subsection = is_subsection
 
     def __repr__(self):
-        return "<Section (%d)>" % len(self.body)
+        return "<Section (%s)>" % self.name
```

Comment:
Any chance ever two sections having the same name?

Reply:
I don't think so, no, they are unique by construction. Also, if they ever were, you'd get a compilation error due to duplicate variable names

Reply:
How about explicitly catching duplicates so Devito can handle the error more gracefully rather than relying upon the end compiler.

Reply:
as I wrote above:

>  they are unique by construction

that is, there cannot be duplicates